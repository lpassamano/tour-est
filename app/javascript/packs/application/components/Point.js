import React, { Component } from "react";
import PropTypes from "prop-types";

export class Point extends Component {
  render() {
    const { id, title, image } = this.props.point;
    return (
      <div key={id}>
        {image ? <img src={image} alt="" width="150px" /> : null}
        <h4>{title}</h4>
      </div>
    );
  }
}

Point.propTypes = {
  point: PropTypes.shape({
    title: PropTypes.string.isRequired,
    caption: PropTypes.string,
    description: PropTypes.string,
    location: PropTypes.string,
    directions: PropTypes.string,
    image: PropTypes.string
  }).isRequired
};

export default Point;
