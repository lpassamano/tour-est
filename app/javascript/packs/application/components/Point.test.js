import { Point } from "./Point";
import React from "react";
import { shallow } from "enzyme";

describe("<Point />", () => {
  const setup = () => {
    const point = {
      point: {
        title: "First point"
      }
    };
    const component = shallow(<Point point={point} />);
    return component;
  };

  test("render", () => {
    const component = setup();
    expect(component).toMatchSnapshot();
  });
});
