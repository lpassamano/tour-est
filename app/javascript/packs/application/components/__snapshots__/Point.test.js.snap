// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Point /> render 1`] = `
<div>
  <div
    className="media-object-basics-example"
  >
    <h4 />
    <div />
    <[object Object]
      onClick={[Function]}
      to="#"
    >
      <i
        className="fi-pencil"
      />
       Edit
    </[object Object]>
    <[object Object]
      onClick={[Function]}
      to="#"
    >
      <i
        className="fi-trash"
      />
       Delete
    </[object Object]>
    <hr />
  </div>
</div>
`;
